### PR TITLE
Added configurable options, fixes #2

### LINKED ISSUE #2: Inquirer.js now accepts input/output options
Hi @knownasilya, while I was playing with Inquirer's new input/output streams options (https://github.com/SBoudrias/Inquirer.js/pull/280) I stumbled upon some problems with **cli-width** making direct access to the `process.stdout` stream - As you can see in the file changes from the linked PR, the output stream now can be user defined and not only `process.stdout`. 

Now, I don't want to force any design/api but it would be really nice if the module could accept an object as the function parameter, that let us set the `output` stream (it can always default to `process.stdout`) and even defining the `defaultWidth` value too - I find it quite non-standard the way we define the default value as a property before invoking the function.

Let me know if a PR for these changes is welcome and most importantly, many thanks for all the good work :trophy:

### PR DESCRIPTION
**Pull request by ruyadorno**
@knownasilya here is the PR that changes the API to accept a configurable options parameter instead.

The implementation is based on our exchanged ideas from #2 and I also provided tests, docs and examples on the README file :blush: let me know what you think of it

Like you mentioned in the original issue, this should definitely be a major release as we are introducing breaking changes :+1: overall this is a much more flexible API now

cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,7 +24,43 @@ cliWidth(); // maybe 204 :)
...
+
+Defining both a default width value and a stream output to try to read from:
+
+```js
+var cliWidth = require('cli-width');
+var ttys = require('ttys');
+
+cliWidth({
+    defaultWidth: 80,
+    output: ttys.output
```
Comment (knownasilya) [inline README.md]: Small nit pick. Could you make the indentation in the example code two spaces?

File: README.md [hunk 2]
```diff
@@ -24,7 +24,43 @@ cliWidth(); // maybe 204 :)
...
+```
+
+Defines a different tty module to read width from:
+
+```js
+var cliWidth = require('cli-width');
+var ttys = require('ttys');
+
+cliWidth({
+    tty: ttys
```
Comment (knownasilya) [inline README.md]: Indentation in the code as well. Thanks

Comment (knownasilya) [inline README.md]: I need to add an eslint config

**Comment by knownasilya** [comment]
I left some comments regarding style, but otherwise it looks great.

**Comment by ruyadorno** [comment]
:+1: sure thing!

**Comment by ruyadorno** [comment]
also fixed the indentation on the `index.js` file :+1:

**Comment by knownasilya** [comment]
Excellent

**Comment by ruyadorno** [comment]
:confetti_ball:

**Pull request merged**